R/ebird_conn.R

```diff
@@ -29,25 +31,48 @@
 #' 
 #' unlink(temp_dir, recursive = TRUE)
 ebird_conn <- function(dataset = c("observations", "checklists"), 
+                       cache_connection = TRUE,
                        memory_limit = 4) {
   dataset <- match.arg(dataset)
+  stopifnot(is.logical(cache_connection), length(cache_connection) == 1)
   stopifnot(is.numeric(memory_limit), length(memory_limit) == 1,
             !is.na(memory_limit), memory_limit > 0)
   parquet <- ebird_parquet_files(dataset = dataset)
   
-  ## Is it worth persisting the duckdb connection on disk to avoid
-  ## recreating the View? (~ 9m operation)
+  # query to create view in duckdb to the parquet file
+  view_query <- paste0("CREATE VIEW '", dataset, 
+                       "' AS SELECT * FROM parquet_scan('",
+                       parquet, "');")
+  
+  # check for a cached connection
+  conn <- mget("birddb", envir = birddb_cache, ifnotfound = NA)[["birddb"]]
+  if (inherits(conn, "DBIConnection") ) {
```

Comment:
I agree, it is a bit confusing. Ideally that first `if (inherits(conn, "DBIConnection")) { ... }` would always return, but it's not obvious to me how to restructure this so it does that while also handling the case when `DBI::dbIsValid(conn)` is `FALSE`. If I'd written this myself I never would have thought to include that bit at all, but I copied that check from `arkdb`, so assuming it's necessary?

Also would be nice to avoid identical `DBI::dbSendQuery(conn, view_query)` calls twice, but again not immediately obvious to me how to accomplish that.

Do you have any suggestions for how to rewrite this conditional block?

Reply:
good question, does look tricky.  I'll take a stab at this in a separate PR